elastic/kibana
# [Security Solutions] Refactor search bar to use global query string

original issue https://github.com/elastic/kibana/issues/123617
Following up https://github.com/elastic/kibana/pull/134210/
Documentation [use_url_state proposal](https://docs.google.com/document/d/1KTYy6RTqafnWYL7GtDt2Jdn5HmsheUrM3peeDK84Trs)


## Summary
Refactor search bar to use global query. It migrates `query`, `filters` and `savedQuery`.


What to test
* It should not present changes to the final user.
* Use the Search bar query, filter, and saved query. Refresh the page. The state should be preserved.
* Add pinned filters and check that they are persistent across different apps.
* Unpinned filter from other apps should not show on security.
* When loading saved queries, it should overwrite filters and pinned filters.

### Migration status

- [x] soucerer 
- [x] **app query** [current PR]
- [x] **filters** [current PR]
- [x] **saved query** [current PR]
- [ ] timerange



### Checklist

Delete any items that are not applicable to this PR.
